controllers/array_action/array_mediator_svc.py

```diff
@@ -951,12 +962,23 @@ def _rmsnapshot(self, internal_snapshot_id):
                 raise array_errors.ObjectNotFoundError(internal_snapshot_id)
             raise ex
 
+    def _handle_delete_snapshot(self, snapshot_name, internal_snapshot_id):
+        if self._is_addsnapshot_supported():
+            if snapshot_name:
+                cli_snapshot = self._get_cli_snapshot_by_name(snapshot_name)
+                if cli_snapshot and cli_snapshot.snapshot_name == snapshot_name:
```

Comment:
is there a chance `_get_cli_snapshot_by_name` will return a snapshot with a different name?

Reply:
right, I cleaned the code